develop/addons/components/genericsetup.rst

```diff
@@ -184,33 +167,51 @@ Unit testing example::
     # Run the extended profile of the betahaus.emaillogin package.
     setup_tool.runAllImportStepsFromProfile('profile-betahaus.emaillogin:extended')
 
+.. note::
+
+    Since Products.GenericSetup 1.8.0 (Plone 4.3.8, Plone 5.0.0) the ``profile-`` part is optional.
+    The code can handle both.
+
 
 Missing upgrade procedure
 =========================
 
-In the add-ons control panel you may see a warning that your add-on is
-`missing an upgrade procedure <http://stackoverflow.com/questions/15316583/how-to-define-a-procedure-to-upgrade-an-add-on>`_.
-
+In the add-ons control panel you may see a warning that your add-on is `missing an upgrade procedure <http://stackoverflow.com/questions/15316583/how-to-define-a-procedure-to-upgrade-an-add-on>`_.
 This means you need to write some `Upgrade steps`_.
 
 
 Uninstall profile
 =================
 
-For the theory, see:
-`<http://blog.keul.it/2013/05/how-to-make-your-plone-add-on-products.html>`_
+For the theory, see `<http://blog.keul.it/2013/05/how-to-make-your-plone-add-on-products.html>`_
+
+For an example, see the `collective.pdfpeek source code <https://github.com/collective/collective.pdfpeek/tree/master/collective/pdfpeek/profiles>`_.
+
+Uninstalling has various parts:
+
+1. CMFQuickInstaller keeps a list of *some* items that were added during install of an add-on, for example css files in the registry.
+   During uninstall, these items are automatically removed.
+   This works okay, but it only undoes a subset of the changes made by the add-on.
+   There have been problems where items were removed that should not have been removed, especially when add-on authors were not aware of this automatic removal.
+   Best practice is to write your own uninstall code.
+
+2. CMFQuickInstaller looks in ``Extensions/install.py`` for an uninstall method and runs it.
+   Various add-ons have added an uninstall method and simply let it apply an ``uninstall`` profile.
+
+3. CMFQuickInstaller 3.0.9 (Plone 4.3.7 and higher) makes the second part a bit easier.
+   If the uninstall method is not found, it looks for a profile with id ``uninstall`` and applies it.
+
+.. note::
 
-For an example, see the `collective.pdfpeek source code
-<https://github.com/collective/collective.pdfpeek/tree/master/collective/pdfpeek/profiles>`_.
+    Plone 5.1 is scheduled to not use the CMFQuickInstaller, but always apply an ``uninstall`` profile.
+    For more information on this plan,
+    see `PLIP 1340 <https://github.com/plone/Products.CMFPlone/issues/1340>`_.
 
```

Comment:
We will have to update that, when we will release 5.1, seeing that is it handy to write it like that ? Maybe we should already prepare a branch where we have done this for 5.1 ? 
I fear if we do not prepare that now, we will forget it.